game-core/src/main/java/org/triplea/performance/PerfTimer.java

```diff
@@ -23,14 +23,16 @@
 @Log
 public class PerfTimer implements Closeable {
 
-  private static final PerfTimer DISABLED_TIMER = new PerfTimer("disabled");
   private static final Map<String, AtomicLong> runningTotal = new HashMap<>();
-  final String title;
+  private static final Map<String, AtomicLong> runningCount = new HashMap<>();
+  private final String title;
   private final long startMillis;
+  private final int reportingFrequency;
 
-  private PerfTimer(final String title) {
+  private PerfTimer(final String title, final int reportingFrequency) {
     this.title = title;
-    this.startMillis = System.nanoTime();
+    this.reportingFrequency = reportingFrequency;
+    this.startMillis = this.reportingFrequency > 0 ? System.nanoTime() : 0;
```

Comment:
Start millis being either the reporting frequency or the actual start nano time is potentially confusing at two levels.

1. The variable says milli though we assign nanos to it.
2. The variable probably should be one value or another. Having a variable's meaning depend on the context and previous call history is a SRP violation. Perhaps we really want two different variables and make them internally nullable. Another option is to have a simple internal object to track this for us, that object could then provide a nice API to abstract away time keeping details.

Reply:
> Start millis being either the reporting frequency or the actual start nano time is potentially confusing at two levels.
> 
> 1. The variable says milli though we assign nanos to it.

Yep, that was wrong in the existing code! Fixed.

> 2. The variable probably should be one value or another. Having a variable's meaning depend on the context and previous call history is a SRP violation. Perhaps we really want two different variables and make them internally nullable. Another option is to have a simple internal object to track this for us, that object could then provide a nice API to abstract away time keeping details.

Just to clarify, this wasn't storing the reporting frequency. It was just conditionally storing the nanoTime if the reporting frequency was > 0. I've switched to using Optional<> to make that clearer and changed from a ternary operator to an if with a comment.